src/bit-systems/audio-emitter-system.ts

```diff
@@ -0,0 +1,113 @@
+import { addComponent, addEntity, defineQuery, removeComponent } from "bitecs";
+import { PositionalAudio, Audio as StereoAudio, AudioListener as ThreeAudioListener } from "three";
+import { HubsWorld } from "../app";
+import { AudioEmitter, AudioSettingsChanged } from "../bit-components";
+import { AudioType, SourceType } from "../components/audio-params";
+import { AudioSystem } from "../systems/audio-system";
+import { applySettings, getCurrentAudioSettings, updateAudioSettings } from "../update-audio-settings";
+import { addObject3DComponent, swapObject3DComponent } from "../utils/jsx-entity";
+
+export type AudioObject3D = StereoAudio | PositionalAudio;
+type AudioConstructor<T> = new (listener: ThreeAudioListener) => T;
+
+export const Emitter2Audio = (AudioEmitter as any).audios as Map<number, number>;
+export const Emitter2Params = (AudioEmitter as any).params as Map<number, number>;
+
+export const EMITTER_FLAGS = {
+  MUTED: 1 << 0,
+  PAUSED: 1 << 1,
+  CLIPPED: 1 << 2
+};
+
+export function isPositionalAudio(node: AudioObject3D): node is PositionalAudio {
+  return (node as any).panner !== undefined;
+}
+
+export function cleanupAudio(audio: AudioObject3D) {
+  const eid = audio.eid!;
+  audio.disconnect();
+  const audioSystem = APP.scene?.systems["hubs-systems"].audioSystem;
+  APP.audios.delete(eid);
+  APP.supplementaryAttenuation.delete(eid);
+  APP.audioOverrides.delete(eid);
+  audioSystem.removeAudio({ node: audio });
+}
+
+function swapAudioType<T extends AudioObject3D>(
+  world: HubsWorld,
+  audioSystem: AudioSystem,
+  eid: number,
+  NewType: AudioConstructor<T>
+) {
+  const audio = world.eid2obj.get(eid)! as AudioObject3D;
+  audio.disconnect();
+  APP.gainMultipliers.delete(eid);
+  APP.sourceType.set(eid, SourceType.MEDIA_VIDEO);
+  APP.supplementaryAttenuation.delete(eid);
+  APP.audioOverrides.delete(eid);
```

Comment:
I don't think `audioOverrides` should be deleted when swapping audio types, should they?

Reply:
Same, I think this should stay intact.